src/Writers/HTMLWriter.jl

```diff
@@ -598,17 +598,17 @@ the page also happens to be a level 1 heading. In that case it is assumed to be
 the page title and is dropped from the list of subsections.
 """
 function collect_subsections(page::Documents.Page)
-    hs = filter(enumerate(page.elements)) do _
-        idx, element = _
-        isa(element, Base.Markdown.Header) || return false # ignore non-headers
-        (idx == 1) && (Utilities.header_level(element) == 1) && return false # if first elem. <h1> => ignored
-        Utilities.header_level(element) <= 2 # only let <h1> and <h2> through
-    end
-    map(hs) do _
-        idx, heading = _
-        anchor = page.mapping[heading]
-        (Utilities.header_level(heading) == 1), "#$(anchor.id)-$(anchor.nth)", heading.text
+    local sections = []
+    for element in page.elements
+        if isa(element, Base.Markdown.Header) && Utilities.header_level(element) < 3
+            local toplevel = Utilities.header_level(element) === 1
```

Comment:
Can break in other cases though (`BigInt(1) === BigInt(1)`). But I guess it comes down to a judgement call, specific to the situation. In this case, since we know we should be dealing with bitstypes, being more strict is probably better, as you pointed out.